src/Serializer/AbstractItemNormalizer.php

```diff
@@ -292,10 +292,28 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
         $propertyNames = $this->propertyNameCollectionFactory->create($context['resource_class'], $options);
 
         $allowedAttributes = [];
+        $groups = $context[self::GROUPS] ?? false;
+        $attributesInGroup = [];
+
+        if ($groups && !\is_array($groups)) {
+            $groups = [$groups];
+        }
+
+        if (null !== $this->classMetadataFactory && $groups) {
+            foreach ($this->classMetadataFactory->getMetadataFor($classOrObject)->getAttributesMetadata() as $attributeMetadata) {
+                $name = $attributeMetadata->getName();
+
+                if (array_intersect($attributeMetadata->getGroups(), $groups)) {
```

Comment:
IIRC, Symfony we have an helper method to do that.

Reply:
You recall correctly. The [getProperties method of Symfonys SerializerExtractor][1] handles this. However, I explained [in issue #2500][2] why we need an additional check. 

  [1]: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/PropertyInfo/Extractor/SerializerExtractor.php#L36
  [2]: https://github.com/api-platform/core/issues/2500#issuecomment-462069294